### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by Sirrah**
I've updated the example with the latest dependencies. Including the build tools, support libs and more relevant Realm 2.2.1.

This also replaces the com.neenbedankt.android-apt plugin with the annotation processor included in the recent Android build tools.

I also did some additional cleanup, removed unused variables etc.

### CODE REVIEW

**Comment by Zhuinden** [comment]
Really cool PR!

The reason why I'm not merging it yet is because I don't want this example to be updated to Realm 2.x yet, until the following are fixed / merged:

- https://github.com/realm/realm-java/issues/3735 and https://github.com/realm/realm-java/issues/3858

- https://github.com/realm/realm-java/issues/3802

- https://github.com/realm/realm-java/issues/3804 (optional)

- https://github.com/realm/realm-java/issues/3651

- https://github.com/realm/realm-java/pull/3834

---------

Basically:

- ObjectStore Results integration (because I think it fixes a bunch of their current bugs)

- BlackBerry compatibility fix

- Samsung Tab 3 Lite 8 fix

- maybe fine-grained notification after Results integration because of https://github.com/realm/realm-java/issues/3427

--------

Until then, I'm keeping this example at 1.2.0

**Comment by Zhuinden** [comment]
Referenced in issue https://github.com/Zhuinden/realm-book-example/issues/8

**Comment by Sirrah** [comment]
Thanks, I'd like those bugs fixed as well. Do consider that having an example Realm app available could help with tracking these bugs down ;)

**Comment by Zhuinden** [comment]
@Sirrah there's actually a [Sync Realm App](https://github.com/realm/RealmTasks/tree/master/RealmTasks%20Android/app/src/main/java/io/realm/realmtasks) but I find it remarkably confusing. Even the UI.

**Comment by Zhuinden** [comment]
Thank you for your contribution, Realm version updated to 3.1.1
